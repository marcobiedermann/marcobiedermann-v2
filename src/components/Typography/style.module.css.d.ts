export const typographyTextAlignCenter: string;
export const typographyTextAlignLeft: string;
export const typographyTextAlignRight: string;
