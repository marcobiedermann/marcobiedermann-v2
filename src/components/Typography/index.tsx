import classNames from 'classnames';
import camelCase from 'lodash/camelCase';
import capitalize from 'lodash/capitalize';
import React, { FC } from 'react';
import styles from './style.module.css';

type Component = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p';
type TextAlign = 'center' | 'left' | 'right';
type TextAlignClass = 'typographyTextAlignCenter' | 'typographyTextAlignLeft' | 'typographyTextAlignRight';

export interface TypographyProps {
  className?: string;
  textAlign?: TextAlign;
}

export const Typography: FC<TypographyProps> = (props) => {
  const { className, textAlign, ...otherProps } = props;

  return (
    <div
      className={classNames(className, {
        [styles[camelCase(`typographyTextAlign${capitalize(textAlign)}`) as TextAlignClass]]: textAlign,
      })}
      {...otherProps}
    />
  );
};

export default Typography;
