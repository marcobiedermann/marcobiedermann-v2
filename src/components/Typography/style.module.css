.typography--text-align-center {
  text-align: center;
}

.typography--text-align-left {
  text-align: left;
}

.typography--text-align-right {
  text-align: right;
}
