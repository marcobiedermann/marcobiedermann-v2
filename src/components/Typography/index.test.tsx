/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import renderer from 'react-test-renderer';
import Typography from '.';

describe('Typography component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Typography />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should align text to left', () => {
    const tree = renderer
      .create(
        <Typography textAlign="left">
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Nemo expedita soluta quis, eveniet, fuga iusto
            similique ut est, molestiae velit illum. Commodi ab obcaecati consectetur eos molestiae assumenda earum
            magni.
          </p>
        </Typography>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should align text to center', () => {
    const tree = renderer
      .create(
        <Typography textAlign="center">
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Nemo expedita soluta quis, eveniet, fuga iusto
            similique ut est, molestiae velit illum. Commodi ab obcaecati consectetur eos molestiae assumenda earum
            magni.
          </p>
        </Typography>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should align text to right', () => {
    const tree = renderer
      .create(
        <Typography textAlign="right">
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Nemo expedita soluta quis, eveniet, fuga iusto
            similique ut est, molestiae velit illum. Commodi ab obcaecati consectetur eos molestiae assumenda earum
            magni.
          </p>
        </Typography>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
