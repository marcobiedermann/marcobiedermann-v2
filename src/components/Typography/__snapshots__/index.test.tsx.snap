// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography component should align text to center 1`] = `
<div
  className="typographyTextAlignCenter"
>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Nemo expedita soluta quis, eveniet, fuga iusto similique ut est, molestiae velit illum. Commodi ab obcaecati consectetur eos molestiae assumenda earum magni.
  </p>
</div>
`;

exports[`Typography component should align text to left 1`] = `
<div
  className="typographyTextAlignLeft"
>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Nemo expedita soluta quis, eveniet, fuga iusto similique ut est, molestiae velit illum. Commodi ab obcaecati consectetur eos molestiae assumenda earum magni.
  </p>
</div>
`;

exports[`Typography component should align text to right 1`] = `
<div
  className="typographyTextAlignRight"
>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Nemo expedita soluta quis, eveniet, fuga iusto similique ut est, molestiae velit illum. Commodi ab obcaecati consectetur eos molestiae assumenda earum magni.
  </p>
</div>
`;

exports[`Typography component should render correctly 1`] = `
<div
  className=""
/>
`;
